### ISSUE TITLE
Release 3.6.0

### ISSUE THREAD

**Issue by niksedk** [SubtitleEdit/subtitleedit#4754]
Time for a new release :)

This release contains a lot of code/project restructuring to ensure SE will also work on .NET in the future (like .NET 5).

SE now uses .NET framework 4.7.2 (has been updated from 4.0). There are many reasons for this: makes much stuff much easier to code, works on recent Linux versions, RichTextBox, security fixes, performance. The downside is that WinXP/Vista will no run SE 3.6+, but SE 3.5.18 is a very good and stable release.

The "source view" is now hidden by default which gives a little more space for the list view. 
It's possible to toggle between "source view" and "list view" by "F2" (or context menu).

Some of the new stuff:
* Edit of bdsup/bdnxml/ts-w-img files
* Dark theme (finally out of beta - OmrSi also made many nice fixes here)
* Textbox syntax coloring + live spell check in text box (much of which was coded by OmrSi)
* Internal store for ASSA styles - thx Leon/z3us
* Copy-paste translate (pretty fast once you get the hang of it - probably works with all translation sites)

Change log: https://raw.githubusercontent.com/SubtitleEdit/subtitleedit/master/Changelog.txt

Beta link: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.18/SubtitleEditBeta.zip

Do let us know if you known/find any bugs :)

TODO:
- [x] Test text boxes
- [x] Test ASS style manager
- [x] Test dark theme
- [x] Test networking
- [x] Test Batch convert
- [x] Test image export
- [x] Test with latest libmpv
- [x] Test OCR
- [x] Test translate
- [x] General testing  (+ fixing bugs)

**Comment by m0ck69** [comment]
I cannot type Polish "ć" letter (alt+c) in text box. Capital "Ć" works tough.

**Comment by moob158** [comment]
@niksedk 

pt-PT language update
[pt-PT.zip](https://github.com/SubtitleEdit/subtitleedit/files/5867670/pt-PT.zip)

**Comment by OmrSi** [comment]
@m0ck69 Check your shortcuts, it might be colliding with one of them.

**Comment by m0ck69** [comment]
@OmrSi nope, that's not it. Everything was OK before but after I updated SE to the newest beta the problem occurred.

**Comment by OmrSi** [comment]
https://github.com/SubtitleEdit/subtitleedit/issues/4500#issuecomment-731059255 I guess this is back.

**Comment by OmrSi** [comment]
> @OmrSi nope, that's not it. Everything was OK before but after I updated SE to the latest beta the problem occurred.
> I can type this letter for instance here but not in text box.
> ![Clipboard01](https://user-images.githubusercontent.com/46628545/105745627-c608ab00-5f3e-11eb-9e4c-2b0215783f8b.jpg)

This is a normal textbox, you're probably using the richtextbox that has syntax coloring in the main window.

**Comment by m0ck69** [comment]
Idk what richtextbox is. I didn't do anything in settings or shortcuts etc. Just updated SE to the beta version and that's it.

**Comment by OmrSi** [comment]
> Idk what richtextbox is. I didn't do anything in settings or shortcuts etc. Just updated SE to the beta version and that's it.

![image](https://user-images.githubusercontent.com/20923700/105747647-b773c100-5f49-11eb-931d-a1b0774d444b.png)

**Comment by m0ck69** [comment]
I don't have it enabled.
![Clipboard01](https://user-images.githubusercontent.com/46628545/105748302-3ebd3680-5f42-11eb-9036-c29e48cf2ccf.jpg)

But I enabled this option for testing and it didn't help either.

**Comment by maknol** [comment]
Is it posible to add option when start a program, to start with this panels in docking and undoking mode:

<a href="http://www.imagebam.com/image/9e73161367951334" target="_blank"><img src="https://thumbs2.imagebam.com/66/2c/88/9e73161367951334.jpg" alt="imagebam.com"></a>

<a href="http://www.imagebam.com/image/12e10d1367951338" target="_blank"><img src="https://thumbs2.imagebam.com/8b/70/cf/12e10d1367951338.jpg" alt="imagebam.com"></a>

It will be nice, because not avery one working with them.

**Comment by niksedk** [comment]
@m0ck69: thx for the info :)
Should be fixed in latest beta: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.18/SubtitleEditBeta.zip
(was due to a typo in this commit: 64f90d718042eb12b33dc5bd71644ba5f4764aa8 )

**Comment by niksedk** [comment]
@maknol: Please make a new issue for this... no new features until after 3.6.0 :)

**Comment by OmrSi** [comment]
@maknol For now, you can make the waveform hidden by default by hiding it from the toolbar.

**Comment by maknol** [comment]
For last 2 version of SE, my Windows(10) gives me that virus alert, and it's only for SE, and yes, including 3.6.0, for old ones - no.

<a href="http://www.imagebam.com/image/c412181368008345" target="_blank"><img src="https://thumbs2.imagebam.com/3e/d0/a1/c412181368008345.jpg" alt="imagebam.com"></a>

**Comment by niksedk** [comment]
Please report this as a false positive to your anti virus provider: https://docs.microsoft.com/en-us/windows/security/threat-protection/microsoft-defender-antivirus/antivirus-false-positives-negatives ... or what provider you have.

Also, Virus Total is a very good tool: Just scanned latest beta: https://www.virustotal.com/gui/file/7af4a960325a14e8d90e00a3634cd967cd90b056f58f1a27ea7c958c8e9d02c8/detection

**Comment by maknol** [comment]
@niksedk for me is not a problem, but someone may find it helpful.

**Comment by niksedk** [comment]
OK, but If anyone are unsure about SE - do try https://www.virustotal.com/gui/

**Comment by maknol** [comment]
@niksedk It's clean, I just shared what Windows give me.

**Comment by shag00** [comment]
I wonder if I might ask if you could alter your naming convention:
"Beta link: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.18/SubtitleEditBeta.zip"

My first reason is that when running 2 instances of SE for the purpose of comparison they both display, in this case, 3.5.18. My second reason is that it is correct to label the existing 3.5.18 beta in the above link as 3.6.0 beta as that is what it is. Another way of looking at it is that beta versions come before release versions, not after.

So far the version you have linked, regardless of it's name, does work on Linux although I have not tested any of the new features.

**Comment by niksedk** [comment]
Naming convention: SE beta versions are called "old version number" + NEXT, beta "number of commits"... 

So `Subtitle Edit 3.5.18 NEXT, beta 715` is the version after 3.5.18, 715 commits ahead of 3.5.18 release.

**Comment by shag00** [comment]
I see the logic, any chance of naming them that way too?

**Comment by m0ck69** [comment]
@niksedk thx for the fix. Now everything is working fine. I can type this letter in both syntax coloring and without it.

**Comment by OmrSi** [comment]
@niksedk ASSA style import doesn't work...

**Comment by niksedk** [comment]
> 
> 
> @niksedk ASSA style import doesn't work...
> 
> EDIT: The format isn't recognized because IsMine is returning false,
> the same issue is causing a problem in overwriting existing files when exporting.
> 
> EDIT2: It looks like the exporting is what's causing the issue, it's not appending a dialog line.
> 
> EDIT3: Fixed it.

ASS style manager had multiple bugs... beta updated: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.18/SubtitleEditBeta.zip

**Comment by OmrSi** [comment]
Yeah, I guess I'll have to update my PR...

**Comment by OmrSi** [comment]
@niksedk Issue in setting line start using Shift + mouse click if the is an original subtitle opened.
If you have two lines like this:
![image](https://user-images.githubusercontent.com/20923700/106345308-de1e5880-62b7-11eb-87ea-73591a7b11be.png)

And you use Shift + mouse click to set the time like this:
![image](https://user-images.githubusercontent.com/20923700/106345320-f9896380-62b7-11eb-9833-eed78cb2da46.png)

The original gets messed up:
![image](https://user-images.githubusercontent.com/20923700/106345336-14f46e80-62b8-11eb-89aa-c342d28d3373.png)

So, steps to recreate:
1. Open a subtitle with an original file.
2. Go to any consecutive 2 lines.
3. Set the start of the first line close to the start of the second.
4. Check out the original subtitle, you'll see that the second line is affected as well, and merging the two lines will cause the second line to disappear.

**Comment by OmrSi** [comment]
Another thing, why doesn't "Apply min. gap between subtitles" work on the original as well?

**Comment by niksedk** [comment]
> 
> 
> @niksedk Issue in setting line start using Shift + mouse click if the is an original subtitle opened.

Good catch :)
Seems to be an older bug... (the ui code around this is not pretty...)

**Comment by OmrSi** [comment]
@niksedk Now when you change a line with Shift + Mouse click, it returns to the old timing when you go to the next line.

**Comment by nkh0472** [comment]
There is a problem between software and Windows font rendering (or GDI Scaling?). I need to right click on the application icon and click "Properties"; go over to the Compatibility tab, enable “Override High DPI scaling behavior” and set the Scaling Behavior to “Application”.
This problem also existed in previous versions.

![before Compatibility setting](https://user-images.githubusercontent.com/67589323/106378635-64828980-63e1-11eb-934e-f08e7c193492.png)
Before compatibility setting ↑

![after Compatibility setting](https://user-images.githubusercontent.com/67589323/106378638-6c422e00-63e1-11eb-8365-792be5356ea5.png)
After compatibility setting ↑

**Comment by OmrSi** [comment]
@niksedk In main, the method `MakeHistoryForUndoOnlyIfNotResent` should be `MakeHistoryForUndoOnlyIfNotRecent` (typo)

And I can't use Undo for using shift + mouse click, it doesn't undo setting the start time.

**Comment by niksedk** [comment]
@nkh0472: I'm not sure what the problem is exactly? The "Before compatibility setting" looks more correct with button sizes etc... I don't know how to test this (I have a standard 1920x1080 pixel monitor running Win10).
Also, probably not something for 3.6.0.

Beta updated: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.18/SubtitleEditBeta.zip

**Comment by nkh0472** [comment]
I'm sorry I didn't describe the problem clearly. 
I am using Windows 10 with 1920x1080 screen resolution with 125% display scaling (which is the recommended scaling by the system for my laptop screen). 
At this point, the font of SubtitleEdit will be a little blurry. It seems that there is something wrong with Windows font rendering (ClearType or something). In this case, I have to turn on the compatibility settings to make the font clear.
The screenshot I uploaded is the beta version of 3.6.0, and the blur is not improved.

**Comment by OmrSi** [comment]
@niksedk https://github.com/SubtitleEdit/subtitleedit/commit/24e9419a86415ee452db15b9ec4cc842dedc04ec#diff-d3c5bcf40e440110e26412b72e525674529b48379351672805f84d6e421e28feR2703
Maybe:  `e.KeyData == UiUtil.GetKeys(Configuration.Settings.Shortcuts.GeneralHelp)`?

**Comment by nkh0472** [comment]
Found another error: #4775

**Comment by niksedk** [comment]
> 
> 
> Found another error: #4775

thx, nice catch and fix :)
Your PR has been merged: https://github.com/SubtitleEdit/subtitleedit/pull/4776

**Comment by PureOcean** [comment]
**File -> Plugins** and **Help -> Update** can't work. Error message:

> The underlying connection was closed: An unexpected error occurred on a receive.

> System: Cannot determine the frame size or a corrupted frame was received.
> 
>    at System.Net.Security._SslStream.EndRead(IAsyncResult asyncResult)
>    at System.Net.TlsStream.EndRead(IAsyncResult asyncResult)
>    at System.Net.Connection.ReadCallback(IAsyncResult asyncResult)

Also, Auto-Translate -> Google-Translate can't work. 

> The request was aborted: Could not create SSL/TLS secure channel.

No all this problem in the stable version (3.5.18.1).

**Comment by niksedk** [comment]
@PureOcean: What OS are you using?

**Comment by nkh0472** [comment]
> At this point, the font of SubtitleEdit will be a little blurry. It seems that there is something wrong with Windows font rendering (ClearType or something). In this case, I have to turn on the compatibility settings to make the font clear.

I find a similar problem here:
https://github.com/paolosalvatori/ServiceBusExplorer/issues/440
And the corresponding commit here:
https://github.com/paolosalvatori/ServiceBusExplorer/pull/362/files

It seems that there are some new features after .Net 4.7 to solve this problem.
And in [High DPI support - Windows Forms .NET Framework | Microsoft Docs](https://docs.microsoft.com/en-us/dotnet/desktop/winforms/high-dpi-support-in-windows-forms), they mentioned some things that need to do to enable these "enhancements for common high DPI and dynamic DPI scenarios".

**Comment by PureOcean** [comment]
> @PureOcean: What OS are you using?

Windows 10.0.19041.746 (64-bit). 

This problem started in the betas in mid-January or early. I wish I hadn't deleted the last working properly beta version. If I could access the old betas, I could test it and tell which version it started on so it would be easy for you to fix. 
How can I now provide you a full debug report?

**Comment by niksedk** [comment]
> 
> 
> > @PureOcean: What OS are you using?
> 
> Windows 10.0.19041.746 (64-bit).
> 
> This issue started in a beta early or mid-January. I wish I hadn't deleted the last working properly beta version. If I could access the old beta version, I could test it and tell which version it started on. So it would be easy for you to fix.
> How can I now provide you a full debug report?

Is this beta any better? https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.18/SubtitleEditBeta.zip

**Comment by PureOcean** [comment]
> Is this beta any better? https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.18/SubtitleEditBeta.zip

I just tried this beta(3.5.18.**760**). The all connection issues is fixed! Thank you very much for the fix and sharing beta build.

**Comment by m0ck69** [comment]
In new beta version the waveform has become very laggy it's not smooth like it was in previous version.

Old version:


https://user-images.githubusercontent.com/46628545/107009508-72475e80-6795-11eb-9e4e-2ac840a111d8.mp4



Beta version:

https://user-images.githubusercontent.com/46628545/107009491-6cea1400-6795-11eb-99d0-5e9845ebcbc9.mp4

**Comment by niksedk** [comment]
> 
> 
> In new beta version the waveform has become very laggy it's not smooth like it was in previous version.
> 
> Old version:
> smooth.mp4
> 
> Beta version:
> laggy.mp4

Are you using mpv as video player? DirectShow? (vlc is not very good at this... until v4)

**Comment by m0ck69** [comment]
Yes, I always use mpv.

**Comment by niksedk** [comment]
@m0ck69: Me too... but I cannot re-create this. Is it the same version of libmpv?  What do you have in the tag `MpvVideoOutputWindows` in `Settings.xml`?

(first time I've seen mp4 inside github - nice :)

**Comment by m0ck69** [comment]
This value seems to be empty.

```
 <VlcLocation />
    <VlcLocationRelative />
    <MpvVideoOutputWindows />
    <MpvVideoOutputLinux>x11</MpvVideoOutputLinux>
    <MpvExtraOption />
    <MpvLogging>False</MpvLogging>
    <MpvHandlesPreviewText>False</MpvHandlesPreviewText>
    <MpvPreviewTextPrimaryColor>White</MpvPreviewTextPrimaryColor>
    <MpvPreviewTextOutlineWidth>1</MpvPreviewTextOutlineWidth>
    <MpvPreviewTextShadowWidth>1</MpvPreviewTextShadowWidth>
    <MpvPreviewTextOpaqueBox>False</MpvPreviewTextOpaqueBox>
    <MpvPreviewTextAlignment>2</MpvPreviewTextAlignment>
    <MpvPreviewTextMarginVertical>10</MpvPreviewTextMarginVertical>
```

**Comment by Flitskikker** [comment]
With the latest beta, the total width of subtitles with multiple lines changes when you click into the textbox.
Looks like with vs. without spaces in between.
Unfortunately, I don't have the time at this moment to look into it myself.

**Comment by niksedk** [comment]
> This value seems to be empty.
Yes, that's actually correct. 
You could try with `direct3d` or `gpu` - see https://mpv.io/manual/master/#video-output-drivers
Sorry, have no idea about this and nor can I re-create this.

**Comment by niksedk** [comment]
> 
> 
> With the latest beta, the total pixel width (bottom right, after Total length) of subtitles with multiple lines changes when you click into the textbox.
> Looks like with vs. without spaces in between.
> Unfortunately, I don't have the time at this moment to look into it myself.

Screenshot?

**Comment by Flitskikker** [comment]
@niksedk 
Let's take this sub for example:
![image](https://user-images.githubusercontent.com/3516155/107119649-4c9b8180-6889-11eb-8dfe-340ff4b8e576.png)
 
 
I select subtitle 1 to check its total length: 8 chars / 114 pixels
![image](https://user-images.githubusercontent.com/3516155/107119674-7785d580-6889-11eb-9a3e-e64e7dfd2408.png)
 
 
But for subtitle 2 it's 8 chars / 123 pixels (it counts a space for concatenating)
![image](https://user-images.githubusercontent.com/3516155/107119691-908e8680-6889-11eb-8eef-27ac14135b62.png)
 
 
But when I click in the edit box, it reverts to 8 chars / 114 pixels:
![image](https://user-images.githubusercontent.com/3516155/107119721-bae04400-6889-11eb-9460-f62e77d4f5bb.png)
 
 
Before I updated to the latest beta (I was on 3.5.16), it showed 8 chars / 114 pixels in both cases.

---
For the "Test OCR" from the TODO, I did some testing with the new nOCR method. I've made a separate issue: #4780

**Comment by niksedk** [comment]
@Flitskikker: thx for the info :)
Should be fixed in latest beta: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.18/SubtitleEditBeta.zip

**Comment by Flitskikker** [comment]
@niksedk 
Thanks for the swift reply!
It now shows "Total length: 8 chars / 65 pixels" for the second subtitle (containing the two lines). 65 pixels is the length of only 1x "test".
It should calculate the total length of the accumulated text, "testtest", which is 114 pixels, as seen in the first subtitle.

**Comment by niksedk** [comment]
> 
> 
> @niksedk
> Thanks for the swift reply!
> It now shows "Total length: 8 chars / 65 pixels" for the second subtitle (containing the two lines). 65 pixels is the length of only 1x "test".
> It should calculate the total length of the accumulated text, "testtest", which is 114 pixels, as seen in the first subtitle.

It now measures the width of the whole text... the longest line in pixels - isn't that's the important info?

**Comment by Flitskikker** [comment]
@niksedk
The idea when I implemented it, is to quickly see if the text could fit on one line.
The same as the total length in characters.

**Comment by niksedk** [comment]
> 
> 
> @niksedk
> The idea when I implemented it, is to quickly see if the text could fit on one line.
> The same as the total length in characters.
> 
> You could try 3.5.16 to see the original behavior.

OK, so it should replace \r\n with space ?

**Comment by Flitskikker** [comment]
@niksedk 
I would replace it with String.Empty like before, and also to be consistent with the total length in characters.
So, that for "test\r\ntest" and "testtest" the total lengths are the same: 8 characters and 114 pixels.

I think, for the total length calculation, we don't replace \r\n with a space, because it's not always needed when changing two lines into one line, for example with Chinese or Japanese.

**Comment by niksedk** [comment]
@Flitskikker: What about adding lengths for each line? Like "93/186" for "Hallo.\r\nHow are you?"

**Comment by Flitskikker** [comment]
@niksedk 
We have that on the left. 😇
![image](https://user-images.githubusercontent.com/3516155/107126989-75387100-68b3-11eb-9fde-ae0da9a6d4a9.png)

**Comment by niksedk** [comment]
> 
> 
> @niksedk
> We have that on the left. 😇
> ![image](https://user-images.githubusercontent.com/3516155/107126989-75387100-68b3-11eb-9fde-ae0da9a6d4a9.png)

Ah, it's "Total length"... my bad!

**Comment by Flitskikker** [comment]
@niksedk 
No worries.

Now that I see those values, I think I have to dive into it anyway...
93+204 = 281? I think I messed up something somewhere.

**Comment by Flitskikker** [comment]
@niksedk 
Fixed calculation and one occurence of an incorrect total width calculation with #4784

**Comment by m0ck69** [comment]
@niksedk I figured out what causes a laggy waveform. It is live spell check. When I turned live spell check off, the waveform became smooth.

**Comment by niksedk** [comment]
> 
> 
> @niksedk
> Fixed pixel width calculation and one occurence of an incorrect total width calculation with #4784

Nice :)
I've optimized the code a little in https://github.com/SubtitleEdit/subtitleedit/commit/4ddd60c3bd11b1b02dea8e6842decf8ce59e66e2 - is it still okay?

**Comment by niksedk** [comment]
> 
> 
> @niksedk I figured out what causes a laggy waveform. It is live spell check. When I turned live spell check off, the waveform became smooth.

Good find :)
We'll have to take a look - perhaps we can fix it, perhaps a feature for next update (or perhaps a comment on the help page...).

**Comment by Flitskikker** [comment]
> Nice :)
> I've optimized the code a little in [4ddd60c](https://github.com/SubtitleEdit/subtitleedit/commit/4ddd60c3bd11b1b02dea8e6842decf8ce59e66e2) - is it still okay?

I deliberately used `TextRenderer.MeasureText` instead of `graphics.MeasureString` because, unfortunately, the latter doesn't give accurate results.
The width will be much higher than it actually is, regardless of setting a different `TextRenderingHint` or `StringFormat`.
More info: https://stackoverflow.com/a/6404811/4489859

For example, this sentence:

> Waarom doet ze zo?
> Als jullie iets hadden, dan moet iemand...

Using `TextRenderer.MeasureText`, the bottom sentence is 575 pixels wide, making it just fit inside the safearea (576 pixels):

![image](https://user-images.githubusercontent.com/3516155/107142635-a7d47f00-6930-11eb-8311-4af42290e240.png)


Using `graphics.MeasureString` it's suddenly a lot wider:

![image](https://user-images.githubusercontent.com/3516155/107142701-f4b85580-6930-11eb-8bd9-be1ec655436c.png)


For reference, rendered in Spot, it's 574 pixels (including border), also just inside the safe area:
![image](https://user-images.githubusercontent.com/3516155/107142749-4fea4800-6931-11eb-9268-454292f11d99.png)

Or in Paint, it's also 575 pixels:
![image](https://user-images.githubusercontent.com/3516155/107142820-b1aab200-6931-11eb-9f2c-f6f24b75420f.png)

I would prefer `TextRenderer.MeasureText` for more accurate results.

**Comment by niksedk** [comment]
> 
> 
> I would prefer `TextRenderer.MeasureText` for more accurate results.

OK, reverted back again in 0bfb97ad8270facd741de16e03f9035a38a38c25
(more than 10 times slower... but correctness is more important and it's still okay fast)

**Comment by Flitskikker** [comment]
@niksedk 
Thank you.
I believe the option to check for line width is disabled by default, so then it doesn't calculate the width every time.

**Comment by niksedk** [comment]
@Flitskikker: thx for fixing and testing :)

Latest changes (incl. many ASS style manager fixes) in latest updated beta: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.18/SubtitleEditBeta.zip

**Comment by OmrSi** [comment]
Hmm, the waveform thing is strange.
It's also not happening here:

With spell check:
![With](https://user-images.githubusercontent.com/20923700/107145872-3d7e0780-694d-11eb-9bcf-a806220028ea.gif)

Without:
![Without](https://user-images.githubusercontent.com/20923700/107145886-55558b80-694d-11eb-9b2c-31daf626b139.gif)

**Comment by OmrSi** [comment]
Also, damn, the dark theme looks cool. :)

**Comment by m0ck69** [comment]
@OmrSi It's the same with all video engines. The lag with live spell check is definitely noticeable here, just like it's shown in the video. When the cursor hits scene hange, itenev  sometimes stopps for a while. Maybe the language is important?

**Comment by OmrSi** [comment]
Can you try using a file with a language for which you have dictionaries?

**Comment by OmrSi** [comment]
@niksedk Deactivate links here as well?
![image](https://user-images.githubusercontent.com/20923700/107150525-da996a00-6966-11eb-9346-c25f1502fdbf.png)

**Comment by OmrSi** [comment]
@m0ck69 Is this any better? [SubtitleEditBeta.zip](https://github.com/SubtitleEdit/subtitleedit/files/5939899/SubtitleEditBeta.zip)

Try it with a saved file and with a changed file and let me know.

**Comment by tomyangsh** [comment]
Hi, could you please try to fix SUP rendering problem on linux version(with mono)? it's completely not working, while simple rendering isn't acceptable for me... Thanks.
![image](https://user-images.githubusercontent.com/20330249/107163708-1b9a7a00-6979-11eb-914e-b49d8ea1fe98.png)

**Comment by m0ck69** [comment]
> 
> 
> @m0ck69 Is this any better? [SubtitleEditBeta.zip](https://github.com/SubtitleEdit/subtitleedit/files/5939899/SubtitleEditBeta.zip)
> 
> Try it with a saved file and with a changed file and let me know.

Now the lag thing is gone!
Thanks so much! Out of curiosity, where was the problem? I noticed that those colorful marks <some text> from text box disappeard. Is it OK?

**Comment by OmrSi** [comment]
@m0ck69 Spell check checks for language change all the time, it never stops, and it was doing that by using the main timer in SE, stopping it until a language check is done. 
I moved the check to another timer and made it so it only checks for language change when the subtitle is changed. 
Now I'm thinking of adding a separate timer for it, that checks every 2 seconds or so. 
I can't think of another way of detecting language change.

**Comment by niksedk** [comment]
OK, beta updated again: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.18/SubtitleEditBeta.zip
Includes OmrSi's fix for "live spell check" waveform lagging  (+ another issue w waveform lagging for moving subs with error)

**Comment by OmrSi** [comment]
@niksedk Shouldn't `e.KeyCode == UiUtil.HelpKeys` be `e.KeyData == UiUtil.HelpKeys`?
`KeyCode` only holds keys, what if the user, like me, uses a modifier with a key? `KeyCode` is needed in that case, right?

**Comment by niksedk** [comment]
> 
> 
> @niksedk Shouldn't `e.KeyCode == UiUtil.HelpKeys` be `e.KeyData == UiUtil.HelpKeys`?
> `KeyCode` only holds keys, what if the user, like me, uses a modifier with a key? `KeyData` is needed in that case, right?

Yes - do you have time or should I fix it?

**Comment by OmrSi** [comment]
It's a simple replacement, I think. I'll do it.
Also, this is way more fun than studying. xD

**Comment by roisag** [comment]
> I cannot type Polish "ć" letter (alt+c) in text box. Capital "Ć" works tough.

Same problem with Polish "ó" (right alt + o). SE 3.5.18 + Win 10 20H2.

**Comment by OmrSi** [comment]
Supposed to be fixed in a later build, have you tried the latest beta?

**Comment by roisag** [comment]
> Supposed to be fixed in a later build, have you tried the latest beta?

Thank you. Fixed in beta.

**Comment by OmrSi** [comment]
@niksedk Found more issues in Live spell check, fixing now.

Live spell check was trying to run in the original text box each time, I think it shouldn't.

Live spell check timer was being disposed of each time the settings is closed even if the live spell check option didn't change.

Right-clicking on the original text box was showing suggestions for the wrong word even if the mouse cursor wasn't on it.

**Comment by uckthis** [comment]
Remove text from HI - Remove text before a colon - only if it is Uppercase, not working in latest beta.

**Comment by niksedk** [comment]
> 
> 
> Remove text from HI - Remove text before a colon - only if it is Uppercase, not working in latest beta.

thx - will have to fix this :)

**Comment by niksedk** [comment]
> 
> 
> Remove text from HI - Remove text before a colon - only if it is Uppercase, not working in latest beta.

Should be fixed in latest beta: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.18/SubtitleEditBeta.zip

**Comment by uckthis** [comment]
@niksedk  Thanks. HI removal works now.

Also, I was testing the invalid italics that you fixed the other day and found another case where it fails.
It's when we have two Italics in a subtitle. Just trying different things to make sure we cover all errors under this.

Here:
[Error.i-2.txt](https://github.com/SubtitleEdit/subtitleedit/files/5977711/Error.i-2.txt)

**Comment by OmrSi** [comment]
@niksedk You should change the organization's image here on GitHub as well...

**Comment by AlexandreMT** [comment]
App is crashing when I try to join subtitle by selecting many files at once. Only works if I pick one by one.

**Comment by niksedk** [comment]
>
>
>AlexandreMT wrote:
> 
> 
> App is crashing when I try to join subtitles by selecting many files at once. Only works if I pick one by one. I'm using latest beta version. Windows 10.

Thx :)
Should be fixed via 7948a36a90890cf2de11776559ef2aa179810069 and in latest beta: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.18/SubtitleEditBeta.zip

**Comment by niksedk** [comment]
SE 3.6.0 out now :)
https://github.com/SubtitleEdit/subtitleedit/releases
